modules/explorer/update.go

```diff
@@ -35,6 +36,14 @@ func (e *Explorer) ReceiveConsensusSetUpdate(cc modules.ConsensusChange) {
 	// Handle incoming blocks
 	for _, block := range cc.AppliedBlocks {
 		e.blockchainHeight += 1
+
+		// Add the current time to seenTimes
+		if time.Unix(int64(block.Timestamp), 0).Before(e.startTime) {
+			e.seenTimes[e.blockchainHeight%types.BlockHeight(len(e.seenTimes))] = time.Unix(int64(block.Timestamp), 0)
+		} else {
+			e.seenTimes[e.blockchainHeight%types.BlockHeight(len(e.seenTimes))] = time.Now()
+		}
```

Comment:
this won't always work because block timestamps can be in the past (only a bit though), but I don't see a better way to do it.

After the explorer has been running for 36+ hours it shouldn't be a problem at all.